test/unit/local/steps/linux_producer.js

```diff
@@ -0,0 +1,184 @@
+/* eslint-env mocha */
+/* @flow */
+
+const should = require('should')
+const os = require('os')
+const fs = require('fs')
+const path = require('path')
+const LinuxProducer = require('../../../../core/local/steps/linux_producer')
+
+if (os.platform() === 'linux') {
+  describe('core/local/steps/linux_producer', () => {
+    describe('API of the producer', () => {
+      let syncPath
+      let producer
+
+      beforeEach(() => {
+        syncPath = fs.mkdtempSync(path.join(os.tmpdir(), 'foo-'))
+        producer = new LinuxProducer({
+          syncPath
+        })
+      })
+
+      it('should register on start and dispose on stop the watcher', async () => {
+        await producer.start()
+        should(producer.watcher).be.not.null()
+        await producer.stop()
+        should(producer.watcher).be.null()
+      })
+
+      it('should change absolute path to relative with process', async () => {
+        const dirname = 'foobar'
+        const batch = [
+          {
+            action: 'created',
+            kind: 'directory',
+            path: path.join(syncPath, dirname)
+          }
+        ]
+        producer.process(batch.map(evt => Object.assign({}, evt)))
+        should(await producer.buffer.pop()).eql(
+          batch.map(evt => Object.assign({}, evt, { path: dirname }))
+        )
+      })
+
+      it('should scan a subfolder tree', async () => {
+        const mypath = ['foo', 'bar', 'baz']
+        mkdirSyncRecursive(syncPath, mypath.join('/'))
+        await producer.start()
+        should(
+          (await producer.buffer.pop()).map(({ action, kind, path }) => ({
+            action,
+            kind,
+            path
+          }))
+        ).eql([
+          {
+            action: 'scan',
+            kind: 'unknown',
+            path: mypath[0]
+          }
+        ])
+        should(
+          (await producer.buffer.pop()).map(({ action, kind, path }) => ({
+            action,
+            kind,
+            path
+          }))
+        ).eql([
+          {
+            action: 'scan',
+            kind: 'unknown',
+            path: `${mypath[0]}/${mypath[1]}`
```

Comment:
We should probably use `path.join` here to make sure paths are correct on Windows as well.

Reply:
I have no plan to test `linux_producer` on Windows, but you are right.

Reply:
You're right, I should have thought about that :man_facepalming: